### PR TITLE
User pref fixes

### PR DESCRIPTION
**Pull request by hanshank**
- delete and get functions for the watchlist are working

### CODE REVIEW

**Comment by glenflorendo** [comment]
- [x] Pull updated staging @hanshank

File: controllers/user.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
 const jwt = require('jsonwebtoken');
-const { Preference, User } = require('../models');
+const { Preference, User, Movie } = require('../models');
+const ObjectId = require('mongodb').ObjectId;
```
Comment (glenflorendo) [inline controllers/user.js]: You don't reference this anywhere.

File: controllers/user.js [hunk 2]
```diff
@@ -17,30 +19,29 @@ function createJWT(user) {
...
-  const newPreference = await new Preference();
-
-  const newUser = await new User({
+  const newPreference = new Preference();
+  const newUser = new User({
     fullName: req.body.fullName,
     email: req.body.email,
     password: req.body.password,
-    preference: newPreference,
+    preferenceId: newPreference._id,
```
Comment (glenflorendo) [inline controllers/user.js:25]: Are you planning to populate this information when Users are queried?

File: controllers/user.js [hunk 3]
```diff
@@ -63,15 +64,49 @@ function login(req, res) {
 }
 
 function watchlist(req, res) {
```
Comment (glenflorendo) [inline controllers/user.js]: Rename to be explicit

File: routes/user.js [hunk 4]
```diff
@@ -4,5 +4,7 @@ const controllers = require('../controllers');
 router.post('/signup', controllers.user.signup);
 router.post('/login', controllers.user.login);
 router.post('/watchlist', controllers.user.watchlist);
```
Comment (glenflorendo) [inline routes/user.js]: This shouldn't be a `POST`. Use URL parameters instead of the request body.

File: routes/user.js [hunk 5]
```diff
@@ -4,5 +4,7 @@ const controllers = require('../controllers');
...
+router.post('/watchlist/movies', controllers.user.getWatchlistMovies);
```
Comment (glenflorendo) [inline routes/user.js]: This shouldn't be a `POST`. Use URL parameters instead of the request body.

File: routes/user.js [hunk 6]
```diff
@@ -4,5 +4,7 @@ const controllers = require('../controllers');
...
+router.delete('/watchlist/movies', controllers.user.deleteWatchlistMovie);
```
Comment (glenflorendo) [inline routes/user.js]: Use URL parameters instead of the request body.

File: controllers/user.js [hunk 7]
```diff
@@ -63,15 +64,49 @@ function login(req, res) {
...
-    return res.status(200).json({message: 'added to watchlist'})
+    .then(res.status(200).json({message: 'added to watchlist'}))
     .catch(err => res.status(401).json(err));
 }
 
+// Gets all the watchlist movies from a user
+const getWatchlistMovies = async (req, res) => {
+  const user = await User.findById(req.body.userId);
+  const movies = await Movie.find({_id: {$in: user.watchlist} })
+  .select('_id images.poster')
```
Comment (glenflorendo) [inline controllers/user.js]: Why don't you want to pass in any other information to the client?

File: controllers/user.js [hunk 8]
```diff
@@ -63,15 +64,49 @@ function login(req, res) {
...
+  .catch(e => {
+    if (e) {
```
Comment (glenflorendo) [inline controllers/user.js]: This is redundant.

File: controllers/user.js [hunk 9]
```diff
@@ -63,15 +64,49 @@ function login(req, res) {
...
+
+  res.status(200).json(movies);
+}
+
+// Deletes a selected watchlist movie
+const deleteWatchlistMovie = (req, res) => {
+  const movieId = req.body.movieId;
+  const userId = req.body.userId
+
+  try {
```
Comment (glenflorendo) [inline controllers/user.js]: You mix `try-catch` and `.then, .catch`.

File: controllers/user.js [hunk 10]
```diff
@@ -63,15 +64,49 @@ function login(req, res) {
...
+    User.findOneAndUpdate(
+      { _id: userId },
+      { $pull: { watchlist: { $in: [movieId] } } },
```
Comment (glenflorendo) [inline controllers/user.js]: Why did you put `movieId` in an array?

File: controllers/user.js [hunk 11]
```diff
@@ -63,15 +64,49 @@ function login(req, res) {
...
+      { multi: true }
```
Comment (glenflorendo) [inline controllers/user.js]: This should be unnecessary, since you treat the watchlist array as a set.

**Review by iylim — APPROVED** [review]
Yay it works!

**Pull request merged**